internal/api/index.go

```diff
@@ -16,6 +16,29 @@ import (
 	"github.com/photoprism/photoprism/pkg/txt"
 )
 
+// GET /api/v1/index
+func IndexingStatus(router *gin.RouterGroup) {
+	router.GET("/index", func(c *gin.Context) {
+		s := Auth(SessionID(c), acl.ResourcePhotos, acl.ActionRead)
+
+		if s.Invalid() {
+			AbortUnauthorized(c)
+			return
+		}
+
+		conf := service.Config()
+
+		if !conf.Settings().Features.Library {
```

Comment:
Checking for `ActionUpdate` (see above) should be sufficient for our use case, you can remove this `if` block then.